tensorflow/python/training/learning_rate_decay_test.py

```diff
@@ -72,6 +73,36 @@ def testVariables(self):
       expected = .1 * 0.96**(100 // 3)
       self.assertAllClose(decayed_lr.eval(), expected, 1e-6)
 
+class ExponentialTimeDecayTest(test_util.TensorFlowTestCase):
+
+  def testDecay(self):
+    initial_lr = 0.1
+    k = 0.5
+    with self.test_session():
+      for step in range(1, 11):
+        lr = learning_rate_decay.exponential_time_decay(initial_lr, step, k)
+        self.assertAllClose(initial_lr*exp(-k*step), lr.eval())
+      
+class InverseDecayTest(test_util.TensorFlowTestCase):
+
+  def testDecay(self):
+    initial_lr = 0.1
+    k = 0.5
+    with self.test_session():
+      for step in range(1, 11):
+        lr = learning_rate_decay.inverse_time_decay(initial_lr, step, k)
+        self.assertAllClose(initial_lr/(1+k*step), lr.eval())
+      
+class StepDecayTest(test_util.TensorFlowTestCase):
+  
+  def testDecay(self):
+    lr = 0.1
+    k = 0.5
+    num_steps = 5
+    with self.test_session():
+      for step in range(1, 11):
+        lr = learning_rate_decay.step_time_decay(lr, step, num_steps, k)
```

Comment:
The reason I was concerned about this function is that you are assuming the 'lr' passed into the function is updated like this test is doing on every call -- the other functions are always passing an initial learning rate and computing the final learning rate based on the step and steps per epoch, so the result doesn't have to be passed back into a later call of the function.  I think keeping the same type of behavior for this function would be ideal.